### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/custom_models.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/custom_models.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/custom_models.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+```
+
+يمكنك أيضًا استخدام أي طريقة أخرى من فئة [`PretrainedConfig`]، مثل [`~PretrainedConfig.push_to_hub`] لتحميل تكوينك مباشرة إلى Hub.
+
+## كتابة نموذج مخصص
+
+الآن بعد أن أصبح لدينا تكوين ResNet، يمكننا المتابعة لكتابة النموذج. في الواقع، سنكتب نموذجين: أحدهما
+يستخرج الميزات المخفية من دفعة من الصور (مثل [`BertModel`]) والآخر مناسب لتصنيف الصور (مثل [`BertForSequenceClassification`]).
+
+كما ذكرنا سابقًا، سنقوم فقط بلف النموذج بشكل فضفاض لجعله بسيطًا في هذا المثال. الشيء الوحيد الذي نحتاج إلى فعله قبل كتابة هذه الفئة هو رسم خريطة بين أنواع الكتل وفئات الكتل الفعلية. ثم يتم تحديد النموذج من التكوين عن طريق تمرير كل شيء إلى فئة `ResNet`:
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:92]:
```suggestion
 كما ذكرنا سابقًا، سنقوم ببناء نموذج مبسط لتسهيل الفهم في هذا المثال. الخطوة الوحيدة المطلوبة قبل كتابة هذه الفئة هي لربط أنواع وحدات البناء بفئات ذات وحدات بناء فعلية. بعد ذلك، يُعرّف النموذج من خلال التكوين عبر تمرير كل شيء إلى فئة `ResNet`:
```

File: docs/source/ar/custom_models.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    def forward(self, tensor, labels=None):
+        logits = self.model(tensor)
+        if labels is not None:
+            loss = torch.nn.cross_entropy(logits, labels)
+            return {"loss": loss, "logits": logits}
+        return {"logits": logits}
+```
+في كلتا الحالتين، لاحظ كيف نرث من `PreTrainedModel` ونستدعي تهيئة الفئة الفائقة باستخدام `config`
+(مثلما تفعل عند كتابة وحدة `torch.nn.Module` عادية). السطر الذي يحدد `config_class` ليس إلزاميًا، ما لم
+ترغب في تسجيل نموذجك باستخدام الفئات التلقائية (راجع القسم الأخير).
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:158]:
```suggestion
في كلتا الحالتين، لاحظ كيف نرث من `PreTrainedModel` ونستدعي مُهيئ الفئة الرئيسية باستخدام `config` (كما تفعل عند إنشاء وحدة `torch.nn.Module` عادية). ليس من الضروري تعريف `config_class` إلا إذا كنت ترغب في تسجيل نموذجك مع الفئات التلقائية (راجع القسم الأخير).
```

File: docs/source/ar/custom_models.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+
+pretrained_model = timm.create_model("resnet50d", pretrained=True)
+resnet50d.model.load_state_dict(pretrained_model.state_dict())
+```
+
+الآن دعونا نرى كيفية التأكد من أنه عند قيامنا بـ [`~PreTrainedModel.save_pretrained`] أو [`~PreTrainedModel.push_to_hub`]، يتم حفظ كود النموذج.
+
+## تسجيل نموذج مع كود مخصص للفئات التلقائية
+
+إذا كنت تكتب مكتبة توسع 🤗 Transformers، فقد ترغب في توسيع الفئات التلقائية لتضمين نموذجك الخاص. يختلف هذا عن دفع الكود إلى Hub بمعنى أن المستخدمين سيحتاجون إلى استيراد مكتبتك للحصول على النماذج المخصصة (على عكس تنزيل كود النموذج تلقائيًا من Hub).
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:192]:
```suggestion
إذا كنت تكتب مكتبة توسع 🤗 Transformers، فقد ترغب في توسيع الفئات التلقائية لتشمل نموذجك الخاص. يختلف هذا عن نشر الكود إلى Hub بمعنى أن المستخدمين سيحتاجون إلى استيراد مكتبتك للحصول على النماذج المخصصة (على عكس تنزيل كود النموذج تلقائيًا من Hub).
```

File: docs/source/ar/custom_models.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+
+ما دام تكوينك يحتوي على سمة `model_type` مختلفة عن أنواع النماذج الموجودة، وأن فئات النموذج لديك لديها السمات الصحيحة `config_class`، فيمكنك ببساطة إضافتها إلى الفئات التلقائية مثل هذا:
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:194]:
```suggestion
ما دام تكوينك يحتوي على معامل  `model_type` مختلفة عن أنواع النماذج الحالية، وأن فئات نماذجك لديك لديها الخصائص الصحيحة `config_class`، فيمكنك ببساطة إضافتها إلى الفئات التلقائية مثل هذا:
```

File: docs/source/ar/custom_models.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+```
+
+بالإضافة إلى أوزان النمذجة والتكوين بتنسيق json، فقد قام هذا أيضًا بنسخ ملفات النمذجة والتكوين `.py` في مجلد `custom-resnet50d` وتحميل النتيجة إلى Hub. يمكنك التحقق من النتيجة في هذا [مستودع النموذج](https://huggingface.co/sgugger/custom-resnet50d).
+
+راجع [البرنامج التعليمي للمشاركة](model_sharing) لمزيد من المعلومات حول طريقة الدفع إلى المحور.
+
+### استخدام نموذج مع كود مخصص
+
+يمكنك استخدام أي تكوين أو نموذج أو مقسم رموز مع ملفات كود مخصص في مستودعه باستخدام الفئات التلقائية و
+طريقة `from_pretrained`. يتم فحص جميع الملفات والرمز المرفوع إلى Hub بحثًا عن البرامج الضارة (راجع وثائق [أمان Hub](https://huggingface.co/docs/hub/security#malware-scanning) لمزيد من المعلومات)، ولكن يجب عليك مراجعة كود النموذج والمؤلف لتجنب تنفيذ التعليمات البرمجية الضارة على جهازك. قم بتعيين `trust_remote_code=True` لاستخدام نموذج مع كود مخصص:
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:314]:
```suggestion
يمكنك استخدام أي تكوين أو نموذج أو مقسم لغوي مع ملفات برمجة مخصصة في مستودعه باستخدام الفئات التلقائية و دالة `from_pretrained`.تُفحص جميع الملفات والرموز المرفوع إلى Hub بحثًا عن البرامج الضارة (راجع وثائق [أمان Hub](https://huggingface.co/docs/hub/security#malware-scanning) لمزيد من المعلومات)، ولكن يجب عليك مراجعة كود النموذج والمؤلف لتجنب تنفيذ التعليمات البرمجية الضارة على جهازك. لتفعيل نموذج يحتوي على شفرة برمجية مخصصة،  عيّن `trust_remote_code=True`:
```

File: docs/source/ar/custom_models.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+
+```py
+from transformers import AutoModelForImageClassification
+
+model = AutoModelForImageClassification.from_pretrained("sgugger/custom-resnet50d", trust_remote_code=True)
+```
+
+يُنصح بشدة أيضًا بتمرير علامة التجزئة للالتزام كـ `revision` للتأكد من أن مؤلف النماذج لم
+قم بتحديث الكود بخطوط جديدة ضارة (ما لم تثق تمامًا بمؤلفي النماذج).
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:323]:
```suggestion
يُنصح بشدة بتحديد رقم إصدار (commit hash) كـ `revision`  للتأكد من عدم تعديل مؤلف النموذج للشفرة لاحقًابإضافة أسطر ضارة  (إلا إذا كنت تثق تمامًا بمؤلفي النموذج):
```

File: docs/source/ar/custom_models.md [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+
+```py
+commit_hash = "ed94a7c6247d8aedce4647f00f20de6875b5b292"
+model = AutoModelForImageClassification.from_pretrained(
+    "sgugger/custom-resnet50d"، trust_remote_code=True، revision=commit_hash
+)
+```
+
+لاحظ أنه عند تصفح سجل الالتزام لمستودع النموذج على Hub، هناك زر لنسخ علامة التجزئة للالتزام بسهولة
+الالتزام.
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:333]:
```suggestion
لاحظ وجود زرّ لنسخ رقم إصدار بسهولة عند تصفح سجل التزامات مستودع النموذج على منصة Hugging Face.
```

File: docs/source/ar/custom_models.md [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+## إرسال الكود إلى Hub
+
+<Tip warning={true}>
+
+هذا API تجريبي وقد يكون له بعض التغييرات الطفيفة في الإصدارات القادمة.
+
+</Tip>
+
+أولاً، تأكد من تعريف نموذجك بالكامل في ملف `.py`. يمكن أن يعتمد على الواردات النسبية إلى بعض الملفات الأخرى طالما أن جميع الملفات موجودة في نفس الدليل (لا ندعم الوحدات الفرعية لهذه الميزة حتى الآن). لمثالنا،
+سنحدد ملف `modeling_resnet.py` وملف `configuration_resnet.py` في مجلد باسم "resnet_model" في دليل العمل الحالي. يحتوي ملف التكوين على كود لـ `ResnetConfig` ويحتوي ملف النمذجة على كود لـ `ResnetModel` و`ResnetModelForImageClassification`.
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:217]:
```suggestion
أولاً، تأكد من تعريف نموذجك بالكامل في ملف `.py`. يمكن أن يعتمد على الاستيراد النسبي لملفات أخرى طالما أن جميع الملفات موجودة في نفس الدليل (لا ندعم الوحدات الفرعية لهذه الميزة حتى الآن). في مثالنا، سنحدد ملف `modeling_resnet.py` وملف `configuration_resnet.py` في مجلد باسم "resnet_model" في دليل العمل الحالي. يحتوي ملف التكوين على كود لـ `ResnetConfig` ويحتوي ملف النمذجة على كود لـ `ResnetModel` و`ResnetModelForImageClassification`.
```

File: docs/source/ar/custom_models.md [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+
+```py
+resnet50d_config = ResnetConfig(block_type="bottleneck", stem_width=32, stem_type="deep", avg_down=True)
+resnet50d = ResnetModelForImageClassification(resnet50d_config)
+
+pretrained_model = timm.create_model("resnet50d", pretrained=True)
+resnet50d.model.load_state_dict(pretrained_model.state_dict())
+```
+
+الآن لإرسال النموذج إلى Hub، تأكد من تسجيل الدخول. إما تشغيل في المحطة الطرفية الخاصة بك:
```
Comment (abodacs) [inline docs/source/ar/custom_models.md:287]:
```suggestion
الآن لإرسال النموذج إلى Hub، تأكد من تسجيل الدخول. إما تشغيل في المحطة الأوامر الطرفية الخاصة بك:
```

**Review by abodacs — CHANGES_REQUESTED** [review]
السﻻم عليكم

@AhmedAlmaghz 
تمت المراجعة جزاء الله خيرا

**Comment by AhmedAlmaghz** [comment]
Thank you @abodacs
